CMakeLists.txt

```diff
@@ -724,6 +724,16 @@ SET(STANDALONE_SOURCE_FILES
     src/ssl.c)
 SET(STANDALONE_COMPILE_FLAGS "-DH2O_USE_LIBUV=0 -DH2O_USE_BROTLI=1 -DQUICLY_USE_TRACER=1")
 IF (WITH_MRUBY)
+    # Bundled mruby cannot be built with Ruby 3.x
+    EXECUTE_PROCESS(
+        COMMAND "${RUBY}" -e "puts RUBY_VERSION"
+        OUTPUT_VARIABLE RUBY_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    IF (NOT ("${RUBY_VERSION}" VERSION_LESS "3.0.0"))
+        MESSAGE(FATAL_ERROR "Ruby 2.x needs for building mruby")
+    ENDIF()
```

Comment:
How about moving this check to where we set `WITH_MRUBY_DEFAULT`?

Otherwise, people trying to build h2o using the default options might see this message without the knowledge of how to turn off the mruby handler.

Reply:
Thanks for comment. I have updated https://github.com/h2o/h2o/pull/2852/commits/c8202aa754c477eaa1637d21a7d6356602ad5455